docker/ftest-docker-containerobject-dsl/src/test/java/org/arquillian/cube/containerobject/dsl/PostgreSqlIT.java

```diff
@@ -0,0 +1,97 @@
+import org.arquillian.cube.docker.impl.client.config.Await;
+import org.arquillian.cube.docker.impl.client.containerobject.dsl.Container;
+import org.arquillian.cube.docker.impl.client.containerobject.dsl.DockerContainer;
+import org.arquillian.cube.docker.impl.requirement.RequiresDocker;
+import org.arquillian.cube.requirement.ArquillianConditionalRunner;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Verify that {@link DockerContainer} can be used to define a valid {@code PosatgreSql} Docker container.
+ * <p>
+ *     A {@link DockerContainer} annotation is used to define a PostgreSql Docker container.
+ *     A local directory is created on the host to hold the PostgreSql data and to be mounted as a volume on the
+ *     Docker container.
+ *     The data directory is created statically, so that it can be used by the {@link DockerContainer} definition later
+ *     in the test lifecycle.
+ *     The {@code Z} bind mode option is passed so that it can be used by SELinux enabled systems, see the related
+ *     <a href="https://docs.docker.com/engine/storage/bind-mounts/#configure-the-selinux-label">Docker documentation</a>,
+ *     otherwise the container would fail to start with the following error:<br>
+ *     {@code mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied}
+ * </p>
+ */
+@Category({ RequiresDocker.class})
+@RunWith(ArquillianConditionalRunner.class)
+public class PostgreSqlIT {
+    static final String POSTGRESQL_USER = "user";
+    static final String POSTGRESQL_PASSWORD = "pass";
+    static final String POSTGRESQL_DATABASE = "test-database";
+    static final String POSTGRESQL_IMAGE = "quay.io/centos7/postgresql-13-centos7:centos7";
+    static final int POSTGRESQL_HOST_PORT_BINDING = 5432;
+    static final String POSTGRESQL_CONTAINER_DATA_DIRECTORY_PATH = "/var/lib/pgsql/data";
+    static final File POSTGRESQL_HOST_DATA_DIRECTORY;
+
+    static {
+        try {
+            POSTGRESQL_HOST_DATA_DIRECTORY = java.nio.file.Files.createTempDirectory(null).toFile();
+        } catch (IOException e) {
+            throw new IllegalStateException("Cannot create PostgreSql data dir", e);
+        }
+    }
+
+    @BeforeClass
+    public static void configurePostgresDataDir() throws IOException {
+        // This is needed to run on SELinux enabled Docker hosts
+        // otherwise the following error will prevent the container from starting successfully:
+        //   mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied
+        java.nio.file.Files.setPosixFilePermissions(Path.of(POSTGRESQL_HOST_DATA_DIRECTORY.toURI()),
```

Comment:
Shall we check the OS before calling `setPosixFilePermissions` in case running on Windows  ?

Reply:
Thanks, fixed.